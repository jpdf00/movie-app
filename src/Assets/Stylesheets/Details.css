.detail {
  margin-top: 50px;
  display: flex;
  flex-direction: column;
  width: 100vw;
  height: auto;
}

.detail__poster {
  border-radius: 5px 5px 0 0;
}

.detail__info {
  display: flex;
  position: absolute;
  width: 100vw;
  bottom: 145px;
  flex-direction: column;
  font-family: 'Montserrat', sans-serif;
  font-size: 12px;
  color: #fff;
  background-color: hsla(360, 100%, 0%, 0.75);
  padding: 10px 10px 15px 10px;
}

.detail__plot {
  display: flex;
  width: 100vw;
  flex-direction: column;
  font-family: 'Montserrat', sans-serif;
  font-size: 10px;
  padding: 10px 10px 15px 10px;
}

.detail__plot__title {
  font-size: 15px;
  font-weight: bold;
  padding: 10px 10px 10px 0;
}

.detail__button {
  display: flex;
  justify-content: center;
  align-items: center;
  color: #fff;
  background-color: #ff6026;
  font-size: 12px;
  height: 60px;
  font-family: 'Montserrat', sans-serif;
}
