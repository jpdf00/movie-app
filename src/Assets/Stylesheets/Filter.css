.filter {
  top: 40px;
  right: 30px;
  position: absolute;
  width: 90px;
  height: 25px;
  font-family: 'Montserrat', sans-serif;
  border-radius: 5px 0 5px 5px;
  border: solid 1px #eee;
  color: #000;
  background-color: #eee;
}
