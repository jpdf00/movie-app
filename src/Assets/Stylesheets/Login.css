.form {
  width: 100vw;
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.form__label {
  padding: 15px;
  font-size: 24px;
  font-family: 'Montserrat', sans-serif;
  text-align: center;
}

.form__input {
  padding: 10px;
  margin: 10px;
  width: 135px;
}

.form__button {
  padding: 10px;
  margin: 10px;
  width: 158px;
  border: 0;
  color: #fff;
  background-color: #ff6026;
}
