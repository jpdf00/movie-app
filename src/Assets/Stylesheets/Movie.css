.movie {
  display: flex;
  flex-direction: column;
  width: 80vw;
  height: auto;
  box-shadow: 2px 2px 15px 3px #aaa;
  margin-bottom: 40px;
}

.movie__poster {
  border-radius: 5px 5px 0 0;

}

.movie__details {
  display: flex;
  flex-direction: column;
  background-color: #fff;
  font-family: 'Montserrat', sans-serif;
  font-size: 12px;
  color: #000;
  padding: 10px 10px 15px 10px;
}

.movie__details__row {
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 10px;
}

.movie__details__row__photo {
  width: 50px;
  height: 50px;
  border-radius: 50px;
  background-image: url('https://m.media-amazon.com/images/M/MV5BMjAwMDI0NTIyOF5BMl5BanBnXkFtZTYwODcyMzk0._V1_.jpg');
  background-position: center;
  background-size: cover;
  margin-right: 5px;
}

.movie__details__row__title {
  font-weight: bold;
}

.movie__details__row__year {
  font-weight: bold;
}

.movie__details__row__stars {
  color: #e3c920;
}

.movie__details__row__runtime {
  color: #aaa;
  font-size: 10px;
}
