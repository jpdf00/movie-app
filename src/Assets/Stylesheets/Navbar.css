.navbar {
  display: flex;
  position: fixed;
  top: 0;
  width: 100vw;
  height: 55px;
  align-items: center;
  justify-content: space-around;
  background-color: #fff;
  padding: 10px;
  font-family: 'Montserrat', sans-serif;
  color: #333;
}

.navbar__menu {
  display: flex;
  flex-direction: column;
  border-radius: 0 5px 5px 5px;
  background-color: #fff;
  top: 40px;
  left: 30px;
  position: absolute;
  padding: 10px 0;
}

.navbar__menu__item {
  padding: 10px 10px;
  border-bottom: solid 1px #333;
  color: #000;
}
