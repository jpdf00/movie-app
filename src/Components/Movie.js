import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
// import '../Assets/Stylesheets/Earth.css';

const Movie = (props) => {
  const { movie, filter } = props;

  return (
    <Link to="\">
      <div className="card">
        <h3 className="topTitle">{movie.title}</h3>

        <p className="topText">
          {
            `Total ${filter}:
            ${movie.year}`
          }
        </p>
      </div>
    </Link>
  );
};

Movie.propTypes = {
  movie: PropTypes.shape({
    ID: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    genre: PropTypes.string.isRequired,
    runtime: PropTypes.number.isRequired,
    year: PropTypes.string.isRequired,
    created_at: PropTypes.string.isRequired,
    updated_at: PropTypes.string.isRequired,
  }).isRequired,
  filter: PropTypes.string.isRequired,
};

export default Movie;
