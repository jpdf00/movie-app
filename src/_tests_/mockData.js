const MOVIES = [{
  id: 3,
  title: 'The Incredible Hulk',
  genre: 'Action',
  runtime: 112,
  year: '2008-06-13',
  created_at: '2021-08-16T16:26:28.275Z',
  updated_at: '2021-08-23T02:26:03.698Z',
  poster: 'https://m.media-amazon.com/images/M/MV5BMTUyNzk3MjA1OF5BMl5BanBnXkFtZTcwMTE1Njg2MQ@@._V1_.jpg',
  director: 'Louis Leterrier',
  photo: 'https://m.media-amazon.com/images/M/MV5BMjAwMDI0NTIyOF5BMl5BanBnXkFtZTYwODcyMzk0._V1_.jpg',
  plot: 'Bruce Banner, a scientist on the run from the U.S. Government, must find a cure for the monster he turns into whenever he loses his temper.',
  favourites: [{
    id: 40, user_id: 1, movie_id: 3, created_at: '2021-08-26T19:29:27.144Z', updated_at: '2021-08-26T19:29:27.144Z',
  }],
}, {
  id: 9, title: 'Thor: The Dark World', genre: 'Action', runtime: 112, year: '2013-11-08', created_at: '2021-08-23T14:01:25.273Z', updated_at: '2021-08-23T14:01:25.273Z', poster: 'https://m.media-amazon.com/images/M/MV5BMTQyNzAwOTUxOF5BMl5BanBnXkFtZTcwMTE0OTc5OQ@@._V1_.jpg', director: 'Alan Taylor', photo: 'https://m.media-amazon.com/images/M/MV5BMjEyMDg3OTgyMl5BMl5BanBnXkFtZTYwNjEyNzIz._V1_.jpg', plot: 'When the Dark Elves attempt to plunge the universe into darkness, Thor must embark on a perilous and personal journey that will reunite him with doctor Jane Foster.', favourites: [],
}, {
  id: 42,
  title: 'Alien',
  genre: 'Horror',
  runtime: 117,
  year: '1979-09-06',
  created_at: '2021-08-26T19:37:07.045Z',
  updated_at: '2021-08-26T19:37:07.045Z',
  poster: 'https://m.media-amazon.com/images/M/MV5BMmQ2MmU3NzktZjAxOC00ZDZhLTk4YzEtMDMyMzcxY2IwMDAyXkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_.jpg',
  director: 'Ridley Scott',
  photo: 'https://m.media-amazon.com/images/M/MV5BMGJkOGM5OWEtNDYxMy00Njg4LWExNjAtY2ZlNWNlNzVhNDk4XkEyXkFqcGdeQXVyNDkzNTM2ODg@._V1_.jpg',
  plot: 'After a space merchant vessel receives an unknown transmission as a distress call, one of the crew is attacked by a mysterious life form and they soon realize that its life cycle has merely begun.',
  favourites: [{
    id: 41, user_id: 1, movie_id: 42, created_at: '2021-08-26T19:38:55.791Z', updated_at: '2021-08-26T19:38:55.791Z',
  }],
}, {
  id: 2,
  title: 'Iron Man',
  genre: 'Action',
  runtime: 126,
  year: '2008-05-02',
  created_at: '2021-08-16T16:25:29.854Z',
  updated_at: '2021-08-23T00:26:21.576Z',
  poster: 'https://m.media-amazon.com/images/M/MV5BMTczNTI2ODUwOF5BMl5BanBnXkFtZTcwMTU0NTIzMw@@._V1_.jpg',
  director: 'Jon Favreau',
  photo: 'https://m.media-amazon.com/images/M/MV5BNjcwNzg4MjktNDNlMC00M2U1LWJmMjgtZTVkMmI4MDI2MTVmXkEyXkFqcGdeQXVyMjI4MDI0NTM@._V1_.jpg',
  plot: 'After being held captive in an Afghan cave, billionaire engineer Tony Stark creates a unique weaponized suit of armor to fight evil.',
  favourites: [{
    id: 1, user_id: 1, movie_id: 2, created_at: '2021-08-23T14:19:43.234Z', updated_at: '2021-08-23T14:19:43.234Z',
  }],
}, {
  id: 4, title: 'Iron Man 2', genre: 'Action', runtime: 125, year: '2010-05-07', created_at: '2021-08-16T16:44:52.103Z', updated_at: '2021-08-23T00:33:42.161Z', poster: 'https://m.media-amazon.com/images/M/MV5BMTM0MDgwNjMyMl5BMl5BanBnXkFtZTcwNTg3NzAzMw@@._V1_.jpg', director: 'Jon Favreau', photo: 'https://m.media-amazon.com/images/M/MV5BNjcwNzg4MjktNDNlMC00M2U1LWJmMjgtZTVkMmI4MDI2MTVmXkEyXkFqcGdeQXVyMjI4MDI0NTM@._V1_.jpg', plot: "With the world now aware of his identity as Iron Man, Tony Stark must contend with both his declining health and a vengeful mad man with ties to his father's legacy.", favourites: [],
}, {
  id: 5, title: 'Thor', genre: 'Action', runtime: 114, year: '2011-05-06', created_at: '2021-08-16T16:46:59.976Z', updated_at: '2021-08-23T00:40:22.399Z', poster: 'https://m.media-amazon.com/images/M/MV5BOGE4NzU1YTAtNzA3Mi00ZTA2LTg2YmYtMDJmMThiMjlkYjg2XkEyXkFqcGdeQXVyNTgzMDMzMTg@._V1_.jpg', director: 'Kenneth Branagh', photo: 'https://m.media-amazon.com/images/M/MV5BMjI0NTQ4Mjk5Ml5BMl5BanBnXkFtZTcwMDc1NjkzNw@@._V1_.jpg', plot: 'The powerful but arrogant god Thor is cast out of Asgard to live amongst humans in Midgard (Earth), where he soon becomes one of their finest defenders.', favourites: [],
}, {
  id: 6, title: 'Captain America: The First Avenger', genre: 'Action', runtime: 124, year: '2011-07-22', created_at: '2021-08-16T16:48:30.515Z', updated_at: '2021-08-23T00:44:24.334Z', poster: 'https://m.media-amazon.com/images/M/MV5BMTYzOTc2NzU3N15BMl5BanBnXkFtZTcwNjY3MDE3NQ@@._V1_.jpg', director: 'Joe Johnston', photo: 'https://m.media-amazon.com/images/M/MV5BNzcxNDQwNzgxNV5BMl5BanBnXkFtZTYwNTQ1MTA0._V1_.jpg', plot: 'Steve Rogers, a rejected military soldier, transforms into Captain America after taking a dose of a "Super-Soldier serum". But being Captain America comes at a price as he attempts to take down a war monger and a terrorist organization.', favourites: [],
}, {
  id: 7,
  title: 'The Avengers',
  genre: 'Action',
  runtime: 143,
  year: '2012-05-04',
  created_at: '2021-08-23T00:50:23.935Z',
  updated_at: '2021-08-23T00:53:33.369Z',
  poster: 'https://m.media-amazon.com/images/M/MV5BNDYxNjQyMjAtNTdiOS00NGYwLWFmNTAtNThmYjU5ZGI2YTI1XkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_.jpg',
  director: 'Joss Whedon',
  photo: 'https://m.media-amazon.com/images/M/MV5BMTg5MzQ0MDA4MF5BMl5BanBnXkFtZTcwNzUwOTk4OQ@@._V1_.jpg',
  plot: "Earth's mightiest heroes must come together and learn to fight as a team if they are going to stop the mischievous Loki and his alien army from enslaving humanity.",
  favourites: [{
    id: 4, user_id: 1, movie_id: 7, created_at: '2021-08-26T16:37:13.700Z', updated_at: '2021-08-26T16:37:13.700Z',
  }],
}, {
  id: 8, title: 'Iron Man 3', genre: 'Action', runtime: 131, year: '2013-05-03', created_at: '2021-08-23T02:22:29.697Z', updated_at: '2021-08-23T02:22:29.697Z', poster: 'https://m.media-amazon.com/images/M/MV5BMjE5MzcyNjk1M15BMl5BanBnXkFtZTcwMjQ4MjcxOQ@@._V1_.jpg', director: 'Shane Black', photo: 'https://m.media-amazon.com/images/M/MV5BMTM2ODI3NzkyNl5BMl5BanBnXkFtZTYwMDQ3OTYz._V1_.jpg', plot: "When Tony Stark's world is torn apart by a formidable terrorist called the Mandarin, he starts an odyssey of rebuilding and retribution.", favourites: [],
}];

const FAVOURITES = [{
  id: 3,
  title: 'The Incredible Hulk',
  genre: 'Action',
  runtime: 112,
  year: '2008-06-13',
  created_at: '2021-08-16T16:26:28.275Z',
  updated_at: '2021-08-23T02:26:03.698Z',
  poster: 'https://m.media-amazon.com/images/M/MV5BMTUyNzk3MjA1OF5BMl5BanBnXkFtZTcwMTE1Njg2MQ@@._V1_.jpg',
  director: 'Louis Leterrier',
  photo: 'https://m.media-amazon.com/images/M/MV5BMjAwMDI0NTIyOF5BMl5BanBnXkFtZTYwODcyMzk0._V1_.jpg',
  plot: 'Bruce Banner, a scientist on the run from the U.S. Government, must find a cure for the monster he turns into whenever he loses his temper.',
  favourites: [{
    id: 40, user_id: 1, movie_id: 3, created_at: '2021-08-26T19:29:27.144Z', updated_at: '2021-08-26T19:29:27.144Z',
  }],
}, {
  id: 9, title: 'Thor: The Dark World', genre: 'Action', runtime: 112, year: '2013-11-08', created_at: '2021-08-23T14:01:25.273Z', updated_at: '2021-08-23T14:01:25.273Z', poster: 'https://m.media-amazon.com/images/M/MV5BMTQyNzAwOTUxOF5BMl5BanBnXkFtZTcwMTE0OTc5OQ@@._V1_.jpg', director: 'Alan Taylor', photo: 'https://m.media-amazon.com/images/M/MV5BMjEyMDg3OTgyMl5BMl5BanBnXkFtZTYwNjEyNzIz._V1_.jpg', plot: 'When the Dark Elves attempt to plunge the universe into darkness, Thor must embark on a perilous and personal journey that will reunite him with doctor Jane Foster.', favourites: [],
}, {
  id: 42,
  title: 'Alien',
  genre: 'Horror',
  runtime: 117,
  year: '1979-09-06',
  created_at: '2021-08-26T19:37:07.045Z',
  updated_at: '2021-08-26T19:37:07.045Z',
  poster: 'https://m.media-amazon.com/images/M/MV5BMmQ2MmU3NzktZjAxOC00ZDZhLTk4YzEtMDMyMzcxY2IwMDAyXkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_.jpg',
  director: 'Ridley Scott',
  photo: 'https://m.media-amazon.com/images/M/MV5BMGJkOGM5OWEtNDYxMy00Njg4LWExNjAtY2ZlNWNlNzVhNDk4XkEyXkFqcGdeQXVyNDkzNTM2ODg@._V1_.jpg',
  plot: 'After a space merchant vessel receives an unknown transmission as a distress call, one of the crew is attacked by a mysterious life form and they soon realize that its life cycle has merely begun.',
  favourites: [{
    id: 41, user_id: 1, movie_id: 42, created_at: '2021-08-26T19:38:55.791Z', updated_at: '2021-08-26T19:38:55.791Z',
  }],
}, {
  id: 2,
  title: 'Iron Man',
  genre: 'Action',
  runtime: 126,
  year: '2008-05-02',
  created_at: '2021-08-16T16:25:29.854Z',
  updated_at: '2021-08-23T00:26:21.576Z',
  poster: 'https://m.media-amazon.com/images/M/MV5BMTczNTI2ODUwOF5BMl5BanBnXkFtZTcwMTU0NTIzMw@@._V1_.jpg',
  director: 'Jon Favreau',
  photo: 'https://m.media-amazon.com/images/M/MV5BNjcwNzg4MjktNDNlMC00M2U1LWJmMjgtZTVkMmI4MDI2MTVmXkEyXkFqcGdeQXVyMjI4MDI0NTM@._V1_.jpg',
  plot: 'After being held captive in an Afghan cave, billionaire engineer Tony Stark creates a unique weaponized suit of armor to fight evil.',
  favourites: [{
    id: 1, user_id: 1, movie_id: 2, created_at: '2021-08-23T14:19:43.234Z', updated_at: '2021-08-23T14:19:43.234Z',
  }],
}, {
  id: 4, title: 'Iron Man 2', genre: 'Action', runtime: 125, year: '2010-05-07', created_at: '2021-08-16T16:44:52.103Z', updated_at: '2021-08-23T00:33:42.161Z', poster: 'https://m.media-amazon.com/images/M/MV5BMTM0MDgwNjMyMl5BMl5BanBnXkFtZTcwNTg3NzAzMw@@._V1_.jpg', director: 'Jon Favreau', photo: 'https://m.media-amazon.com/images/M/MV5BNjcwNzg4MjktNDNlMC00M2U1LWJmMjgtZTVkMmI4MDI2MTVmXkEyXkFqcGdeQXVyMjI4MDI0NTM@._V1_.jpg', plot: "With the world now aware of his identity as Iron Man, Tony Stark must contend with both his declining health and a vengeful mad man with ties to his father's legacy.", favourites: [],
}, {
  id: 5, title: 'Thor', genre: 'Action', runtime: 114, year: '2011-05-06', created_at: '2021-08-16T16:46:59.976Z', updated_at: '2021-08-23T00:40:22.399Z', poster: 'https://m.media-amazon.com/images/M/MV5BOGE4NzU1YTAtNzA3Mi00ZTA2LTg2YmYtMDJmMThiMjlkYjg2XkEyXkFqcGdeQXVyNTgzMDMzMTg@._V1_.jpg', director: 'Kenneth Branagh', photo: 'https://m.media-amazon.com/images/M/MV5BMjI0NTQ4Mjk5Ml5BMl5BanBnXkFtZTcwMDc1NjkzNw@@._V1_.jpg', plot: 'The powerful but arrogant god Thor is cast out of Asgard to live amongst humans in Midgard (Earth), where he soon becomes one of their finest defenders.', favourites: [],
}, {
  id: 6, title: 'Captain America: The First Avenger', genre: 'Action', runtime: 124, year: '2011-07-22', created_at: '2021-08-16T16:48:30.515Z', updated_at: '2021-08-23T00:44:24.334Z', poster: 'https://m.media-amazon.com/images/M/MV5BMTYzOTc2NzU3N15BMl5BanBnXkFtZTcwNjY3MDE3NQ@@._V1_.jpg', director: 'Joe Johnston', photo: 'https://m.media-amazon.com/images/M/MV5BNzcxNDQwNzgxNV5BMl5BanBnXkFtZTYwNTQ1MTA0._V1_.jpg', plot: 'Steve Rogers, a rejected military soldier, transforms into Captain America after taking a dose of a "Super-Soldier serum". But being Captain America comes at a price as he attempts to take down a war monger and a terrorist organization.', favourites: [],
}, {
  id: 7,
  title: 'The Avengers',
  genre: 'Action',
  runtime: 143,
  year: '2012-05-04',
  created_at: '2021-08-23T00:50:23.935Z',
  updated_at: '2021-08-23T00:53:33.369Z',
  poster: 'https://m.media-amazon.com/images/M/MV5BNDYxNjQyMjAtNTdiOS00NGYwLWFmNTAtNThmYjU5ZGI2YTI1XkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_.jpg',
  director: 'Joss Whedon',
  photo: 'https://m.media-amazon.com/images/M/MV5BMTg5MzQ0MDA4MF5BMl5BanBnXkFtZTcwNzUwOTk4OQ@@._V1_.jpg',
  plot: "Earth's mightiest heroes must come together and learn to fight as a team if they are going to stop the mischievous Loki and his alien army from enslaving humanity.",
  favourites: [{
    id: 4, user_id: 1, movie_id: 7, created_at: '2021-08-26T16:37:13.700Z', updated_at: '2021-08-26T16:37:13.700Z',
  }],
}, {
  id: 8, title: 'Iron Man 3', genre: 'Action', runtime: 131, year: '2013-05-03', created_at: '2021-08-23T02:22:29.697Z', updated_at: '2021-08-23T02:22:29.697Z', poster: 'https://m.media-amazon.com/images/M/MV5BMjE5MzcyNjk1M15BMl5BanBnXkFtZTcwMjQ4MjcxOQ@@._V1_.jpg', director: 'Shane Black', photo: 'https://m.media-amazon.com/images/M/MV5BMTM2ODI3NzkyNl5BMl5BanBnXkFtZTYwMDQ3OTYz._V1_.jpg', plot: "When Tony Stark's world is torn apart by a formidable terrorist called the Mandarin, he starts an odyssey of rebuilding and retribution.", favourites: [],
}];

export { MOVIES, FAVOURITES };
