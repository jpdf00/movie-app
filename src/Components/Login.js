import React, { useState } from 'react';
import PropTypes from 'prop-types';
import '../Assets/Stylesheets/Login.css';

const Login = (props) => {
  const { user, handleSubmit } = props;
  const [username, setUsername] = useState(user);

  const handleChange = (e) => {
    setUsername(e.target.value);
  };

  return (
    <form onSubmit={() => handleSubmit(username)} className="form">
      <p className="form__label">To access the site enter your name below:</p>
      <label htmlFor="username">
        <input className="form__input" type="text" name="username" value={username} id="username" placeholder="Enter name here..." onChange={handleChange} />
      </label>
      <button className="form__button" type="submit" name="button">Enter</button>
    </form>
  );
};

Login.propTypes = {
  user: PropTypes.string.isRequired,
  handleSubmit: PropTypes.func.isRequired,
};

export default Login;
