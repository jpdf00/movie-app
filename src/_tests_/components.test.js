import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { MOVIES } from './mockData';
import Login from '../Components/Login';
import Navbar from '../Components/Navbar';
import Movie from '../Components/Movie';
import NotFound from '../Components/NotFound';
import Filter from '../Containers/Filter';
import MoviesList from '../Containers/MoviesList';
import Details from '../Containers/Details';

const initialState = {
  movies: MOVIES,
  movie: MOVIES[0],
  filter: 'All',
  title: 'Movies',
  user: '',
};

const mockFunc = () => true;

const mockStore = configureStore([]);

describe('Login', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore(initialState);

    store.dispatch = jest.fn();

    component = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Login user="user" handleSubmit={mockFunc} />
        </BrowserRouter>
      </Provider>,
    );
  });

  it('Should render', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe('Navbar', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore(initialState);

    store.dispatch = jest.fn();

    component = renderer.create(
      <Provider store={store}>
        <Navbar />
      </Provider>,
    );
  });

  it('Should render', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe('Movie', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore(initialState);

    store.dispatch = jest.fn();

    component = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Movie movie={MOVIES[0]} />
        </BrowserRouter>
      </Provider>,
    );
  });

  it('Should render', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe('Not Found', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore(initialState);

    store.dispatch = jest.fn();

    component = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <NotFound />
        </BrowserRouter>
      </Provider>,
    );
  });

  it('Should render', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe('Filter', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore(initialState);

    store.dispatch = jest.fn();

    component = renderer.create(
      <Provider store={store}>
        <Filter handlechange={mockFunc} />
      </Provider>,
    );
  });

  it('Should render', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe('Movies', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore(initialState);

    store.dispatch = jest.fn();

    component = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <MoviesList path="/list/movies" />
        </BrowserRouter>
      </Provider>,
    );
  });

  it('Should render', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe('Details', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore(initialState);

    store.dispatch = jest.fn();

    component = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Details path="/movies/3" />
        </BrowserRouter>
      </Provider>,
    );
  });

  it('Should render', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
