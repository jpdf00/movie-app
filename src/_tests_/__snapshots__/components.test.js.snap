// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation Should render 1`] = `
<div
  className="form"
>
  <p
    className="form__label"
  >
    You are succesfully loged in.
  </p>
</div>
`;

exports[`Details Should render 1`] = `
<div
  className="processing"
>
  Processing
</div>
`;

exports[`Filter Should render 1`] = `
<select
  className="filter"
  name="filter"
  onChange={[Function]}
  value="All"
>
  <option
    value="All"
  >
    All
  </option>
  <option
    value="Action"
  >
    Action
  </option>
  <option
    value="Biography"
  >
    Biography
  </option>
  <option
    value="Fantasy"
  >
    Fantasy
  </option>
  <option
    value="History"
  >
    History
  </option>
  <option
    value="Horror"
  >
    Horror
  </option>
  <option
    value="Kids"
  >
    Kids
  </option>
  <option
    value="Learning"
  >
    Learning
  </option>
  <option
    value="Sci-Fi"
  >
    Sci-Fi
  </option>
</select>
`;

exports[`Login Should render 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <p
    className="form__label"
  >
    To access the site enter your name below:
  </p>
  <label
    htmlFor="username"
  >
    <input
      className="form__input"
      id="username"
      name="username"
      onChange={[Function]}
      placeholder="Enter name here..."
      type="text"
      value="user"
    />
  </label>
  <button
    className="form__button"
    name="button"
    type="submit"
  >
    Enter
  </button>
</form>
`;

exports[`Movie Should render 1`] = `
<a
  href="/movies/3"
  onClick={[Function]}
>
  <div
    className="movie"
  >
    <img
      alt="Movie Poster"
      className="movie__poster"
      src="https://m.media-amazon.com/images/M/MV5BMTUyNzk3MjA1OF5BMl5BanBnXkFtZTcwMTE1Njg2MQ@@._V1_.jpg"
    />
    <div
      className="movie__details"
    >
      <div
        className="movie__details__row"
      >
        <h3
          className="movie__details__row__title"
        >
          The Incredible Hulk
        </h3>
        <h3
          className="movie__details__row__year"
        >
          2008
        </h3>
      </div>
      <div
        className="movie__details__row"
      >
        <div
          className="movie__details__row__stars"
        >
          <i
            className="fas fa-star"
          />
          <i
            className="fas fa-star"
          />
          <i
            className="fas fa-star"
          />
          <i
            className="fas fa-star"
          />
          <i
            className="fas fa-star"
          />
        </div>
        <p
          className="movie__details__row__runtime"
        >
          112 min
        </p>
      </div>
    </div>
  </div>
</a>
`;

exports[`Movies Should render 1`] = `
<div
  className="processing"
>
  Processing
</div>
`;

exports[`Navbar Should render 1`] = `
<div
  className="navbar"
>
  <i
    className="fas fa-bars"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="menu"
    tabIndex={0}
  />
  <h1
    className="title"
  >
    Movies
  </h1>
  <i
    className="fas fa-search"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="menu"
    tabIndex={0}
  />
</div>
`;

exports[`Not Found Should render 1`] = `
<div
  className="NotFound"
  id="notFound"
/>
`;
